fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java

```diff
@@ -455,30 +464,38 @@ public TTableDescriptor toThrift(List<ReferencedPartitionInfo> partitions) {
         }
 
         // partitions
-        for (ReferencedPartitionInfo info : partitions) {
+        List<PartitionKey> partitionKeys = Lists.newArrayList();
+        for (ReferencedPartitionInfo partition : partitions) {
+            partitionKeys.add(partition.getKey());
+        }
+        List<HivePartition> hivePartitions;
+        try {
+            hivePartitions = getPartitions(partitionKeys);
+        } catch (DdlException e) {
+            LOG.warn("table {} gets partition info failed.", name, e);
+            return null;
+        }
+
+        int i = 0;
+        for (HivePartition hivePartition : hivePartitions) {
+            // HiveRepository.getPartitions future to ensure an orderly
```

Comment:
1. what dose this comment mean?
2. suggest to use for (int i = 0; i < hivePartitions.size(); i++) {